web_programming/get_links_of_url.py

```diff
@@ -0,0 +1,46 @@
+import re
+import urllib.parse
+import urllib.request
+
+from bs4 import BeautifulSoup
+
+
+def is_valid_url(url):
+    """
+    Function Valid the url
+    """
+
+    regex = re.compile(
+        r"^(?:http|ftp)s?://"
+        r"""(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)
+        +(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|"""
+        r"localhost|"
+        r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"
+        r"(?::\d+)?"
+        r"(?:/?|[/?]\S+)$",
+        re.IGNORECASE,
+    )
```

Comment:
That is one complex regex! This needs thorough testing.
Please use atleast 10 ***different*** types of ***correct*** and 10 ***different*** types of ***incorrect*** URLs to test it.

Reply:
This regex is used by the Django framework to validate the URL

Reply:
Oh, so you didn't come up on your own?
That's fine. Please provide the link as a reference for other people.

And we still ***need*** the tests.